pkg/broker/filter/filter_handler.go

```diff
@@ -56,6 +57,12 @@ const (
 	defaultMaxIdleConnectionsPerHost = 100
 )
 
+// HeaderProxyAllowList contains the headers that are proxied from the reply; other than the CloudEvents headers.
+// Other headers are not proxied because of security concerns.
+var HeaderProxyAllowList = map[string]struct{}{
```

Comment:
This is fine. As an FYI, Kubernetes has some helpers here: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#String

There will probably be an equivalent for go 1.19 or so with generics, I expect.